src/python/pants/core/goals/fmt.py

```diff
@@ -342,64 +250,68 @@ async def fmt(
     console: Console,
     specs: Specs,
     fmt_subsystem: FmtSubsystem,
-    build_file_options: BuildFileOptions,
     workspace: Workspace,
     union_membership: UnionMembership,
 ) -> Fmt:
-    fmt_target_request_types, fmt_build_files_request_types = _get_request_types(
-        fmt_subsystem, union_membership
+    fmt_request_types = list(union_membership.get(FmtRequest))
+    target_partitioners = list(union_membership.get(FmtTargetsRequest.PartitionRequest))
+    file_partitioners = list(union_membership.get(FmtFilesRequest.PartitionRequest))
+
+    partitions_by_request_type = await _get_partitions_by_request_type(
+        fmt_request_types,
+        target_partitioners,
+        file_partitioners,
+        fmt_subsystem,
+        specs,
+        lambda request_type: Get(Partitions, FmtTargetsRequest.PartitionRequest, request_type),
+        lambda request_type: Get(Partitions, FmtFilesRequest.PartitionRequest, request_type),
     )
 
-    _get_targets = Get(
-        FilteredTargets,
-        Specs,
-        specs if fmt_target_request_types else Specs.empty(),
-    )
-    _get_specs_paths = Get(
-        SpecsPaths, Specs, specs if fmt_build_files_request_types else Specs.empty()
-    )
+    if not partitions_by_request_type:
+        return Fmt(exit_code=0)
 
-    targets, specs_paths = await MultiGet(_get_targets, _get_specs_paths)
-    specified_build_files = FilespecMatcher(
-        includes=[os.path.join("**", p) for p in build_file_options.patterns],
-        excludes=build_file_options.ignores,
-    ).matches(specs_paths.files)
+    def batch(files: Iterable[str]) -> Iterator[tuple[str, ...]]:
+        batches = partition_sequentially(
+            files,
+            key=lambda x: str(x),
+            size_target=fmt_subsystem.batch_size,
+            size_max=4 * fmt_subsystem.batch_size,
+        )
+        for batch in batches:
+            yield tuple(batch)
+
+    def _make_disjoint_subpartition_batch_requests() -> Iterable[_FmtSubpartitionBatchRequest]:
```

Comment:
I think that adding the docstring(s) or comments referenced in https://github.com/pantsbuild/pants/pull/16735#discussion_r971067249 might still be a good idea to help explain what subpartitions vs batches are at a high level.

Reply:
Still on my TODO list :smile: